### ISSUE TITLE
getBounds() and getTransformedBounds() improvement

### ISSUE THREAD

**Issue by raduanastase** [CreateJS/EaselJS#647]
If you have
```javascript
var bounds = object.getBounds();
var transformedBounds = object.getTransformedBounds();
```
both variables will indicate to the same rectangle and thus will only have the value from `getTransformedBounds`.

In the documentation it's mention that the rectangle returned it's reused internally and if it's needed later on, it should be cloned. 

Wouldn't it be better if these methods would have an optional parameter `createNew` set by default to `false` (to keep backwards compatibility) that would make it return the same rectangle or a new one?

**Comment by gskinner** [comment]
What do you see as the primary advantage of that? It saves a few keystrokes, but isn't a significant accelerator, and it clutters the params which we may need to use later if we get into Shape bound calculations.

```
// current approach:
var bounds = obj.getBounds().clone();

// proposed approach:
var bounds = obj.getBounds(true);
```

**Comment by yar3333** [comment]
@raduanastase It is a known issue. In past, I have a hard-to-find bug with this (getBounds() returns rectangle, which may be changed by EaselJS, so cloning is necessary).

**Comment by gskinner** [comment]
Its not so much a known issue, as a documented compromise for performance. Constructing and destroying new Rectangle instances for these operations can get expensive.

**Comment by raduanastase** [comment]
It's not about saving keystrokes, but being more intuitive for the developer. 

For me it was a problem because:
1. I didn't know what was happening at first. 
2. I realized it was a problem with those two methods after a lot of commented code. 
3. I had to read the documentation and discover the workaround.

If you don't have an IDE with auto completion, no matter how you change them, it wouldn't make a difference, but if you do, you would see the optional parameter and you would know what is happening.

Regarding the cluttering of the params, I didn't thought you would like to use the params for Shape bound calculations. So maybe it's best to keep the current workaround for now. 

Thanks anyway for your input
